jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -559,4 +574,222 @@ public String toString()
                 pooled);
         }
     }
+
+    /** A pluggable strategy to optimize pool acquisition
+     * @param <T> The type of the items in the pool
+     */
+    public interface Cache<T>
+    {
+        /** Acquire an entry
+         * @param entries The list of entries known to the pool. This may be concurrently modified.
+         * @return An acquired entry or null if none can be acquired by this strategy
+         */
+        Pool<T>.Entry acquire(List<Pool<T>.Entry> entries);
+
+        /**
+         * Notification an entry has been release.  The notification comes after the entry
+         * has been put back in the pool and it may already have been reacquired before or during this call.
+         * @param entries The list of entries known to the pool. This may be concurrently modified.
+         * @param entry The entry to be release
+         * @param reusable true if the entry is reusable and will be put back in the pool.
+         */
+        default void released(List<Pool<T>.Entry> entries, Pool<T>.Entry entry, boolean reusable)
```

Comment:
Just `release()` and implementations have to call `entry.tryRelease()` so that we're symmetric with `acquire()` above?

Reply:
I second @sbordet; and wouldn't the following signature be better?

 `void release(Pool<T>.Entry entry);`

without a list of entries (I can't see why it would be needed) and no default impl?